aya/src/programs/uprobe.rs

```diff
@@ -127,38 +127,44 @@ fn resolve_attach_path<T: AsRef<Path>>(
     target: T,
     pid: Option<pid_t>,
 ) -> Result<String, UProbeError> {
-    // Look up the path for the target. If it there is a pid, and the target is a library name that
-    // is in the process's memory map, use the path of that library. Otherwise, use the target
-    // as-is.
+    // Look up the path for the target. If it there is a pid, and the target is a library name
+    // that is in the process's memory map, use the path of that library. Otherwise, use the target as-is.
     let target = target.as_ref();
-    let target_str = &*target.as_os_str().to_string_lossy();
-
-    let mut path = if let Some(pid) = pid {
-        find_lib_in_proc_maps(pid, target_str).map_err(|io_error| UProbeError::FileError {
-            filename: format!("/proc/{pid}/maps"),
-            io_error,
-        })?
-    } else {
-        None
-    };
-
-    if path.is_none() {
-        path = if target.is_absolute() {
-            Some(target_str)
-        } else {
+    let target_str = target.to_str().ok_or(UProbeError::InvalidTarget {
+        path: target.to_owned(),
+    })?;
+
+    let path = pid
+        .and_then(|pid| {
+            find_lib_in_proc_maps(pid, target_str)
+                .map_err(|io_error| UProbeError::FileError {
+                    filename: format!("/proc/{pid}/maps"),
+                    io_error,
+                })
+                .transpose()
+        })
+        .transpose()?;
+    path.or_else(|| {
+        Some(target)
```

Comment:
This expression (`Some().filter().map()`) would be clearer as `target.is_absolute().then(|| target_str.to_owned())`